examples/window_pump_events.rs

```diff
@@ -45,27 +36,25 @@ fn main() {
             match event {
                 Event::WindowEvent {
                     event: WindowEvent::CloseRequested,
-                    ..
-                } => {
-                    quit = true;
-                }
+                    window_id,
+                } if window_id == window.id() => control_flow.set_exit(),
                 Event::MainEventsCleared => {
-                    control_flow.set_exit();
-                }
-                Event::RedrawRequested(_) => {
-                    fill::fill_window(&window);
+                    window.request_redraw();
                 }
                 _ => (),
             }
         });
+        if let PumpStatus::Exit(exit_code) = status {
+            break 'main ExitCode::from(exit_code as u8);
+        }
 
         // Sleep for 1/60 second to simulate rendering
         println!("rendering");
```

Comment:
hehe, I forgot to change that comment to "Update()" to avoid encouraging anyone to render outside of the Winit loop.

I'd probably rather do that, but yeah technically it does work to actually render here too, in Windows, macOS, Linux and Android (we rely on this at Embark currently) but yeah it doesn't align with how macOS expects apps to render in drawRect and I think we probably get artifacts when resizing the window.

Reply:
oh, hah, I squash the change into another patch I have locally:
```rust
--- a/examples/window_pump_events.rs
+++ b/examples/window_pump_events.rs
@@ -48,8 +48,11 @@ fn main() -> std::process::ExitCode {
             break 'main ExitCode::from(exit_code as u8);
         }

-        // Sleep for 1/60 second to simulate rendering
-        println!("rendering");
+        // Sleep for 1/60 second to simulate application work
+        //
+        // Since `pump_events` doesn't block it will be important to
+        // throttle the loop in the app somehow.
+        println!("Update()");
         sleep(Duration::from_millis(16));
     }
 }
```

Reply:
I ended up adding support for filling the window, consistent with how the other examples work, via `RedrawRequested` and changed this comment as with the patch above